test/optimizer/plugins/AddPathRulesOptimizer.test.ts

```diff
@@ -0,0 +1,169 @@
+import optimizer from '../../../src/optimizer/plugins/AddPathRulesOptimizer';
+import {
+  ExportableInstance,
+  ExportableAssignmentRule,
+  ExportablePathRule,
+  ExportableProfile,
+  ExportableFlagRule
+} from '../../../src/exportable';
+import { Package } from '../../../src/processor';
+import { cloneDeep } from 'lodash';
+
+//test for profile too
+//make sure indents are occuring in simplify rule path contexts optimizer
+describe('optimizer', () => {
+  describe('add_rule_paths', () => {
+    let myPackage: Package;
+
+    beforeEach(() => {
+      myPackage = new Package();
+    });
+
+    it('should add path rules based on repeated parent paths', () => {
+      // Instance: ChainsawPatient
+      // InstanceOf: Patient
+      // Usage: #example
+      // * address.city = "Tokyo"
+      // * address.country = "JP"
+      const instance = new ExportableInstance('ChainsawPatient');
+      instance.instanceOf = 'Patient';
+      instance.usage = 'Example';
+      const tokyoCity = new ExportableAssignmentRule('address.city');
+      tokyoCity.value = 'Tokyo';
+      const japanCountry = new ExportableAssignmentRule('address.country');
+      japanCountry.value = 'JP';
+      instance.rules.push(tokyoCity, japanCountry);
+
+      // Instance: ChainsawPatient
+      // InstanceOf: Patient
+      // Usage: #example
+      // * address
+      // * address.city = "Tokyo"
+      // * address.country = "JP"
+      const addressPathRule = new ExportablePathRule('address');
+      const expectedPathRule = cloneDeep(addressPathRule);
```

Comment:
I'm not sure why this rule needs to be cloned. Can we just use `addressPathRule` directly in the expect statement?

Reply:
Agreed. There were some areas where cloning was helpful, but this is not one of them.

Reply:
Tidied up the tests in 5df2936.